cmd/tree-main.go

```diff
@@ -0,0 +1,285 @@
+/*
+ * MinIO Client (C) 2019 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+	"github.com/minio/mc/pkg/probe"
+)
+
+const (
+	treeEntry     = "├─ "
+	treeLastEntry = "└─ "
+	treeNext      = "│"
+	treeLevel     = "  "
+)
+
+// Structured message depending on the type of console.
+type treeMessage struct {
+	Entry        string
+	IsDir        bool
+	BranchString string
+}
+
+// Colorized message for console printing.
+func (t treeMessage) String() string {
+	entryType := "File"
+	if t.IsDir {
+		entryType = "Dir"
+	}
+	return fmt.Sprintf("%s%s", t.BranchString, console.Colorize(entryType, t.Entry))
+}
+
+// JSON'ified message for scripting.
+// Does No-op. JSON requests are redirected to `ls -r --json`
+func (t treeMessage) JSON() string {
+	return ""
+}
+
+var treeFlags = []cli.Flag{
+	cli.BoolFlag{
+		Name:  "files, f",
+		Usage: "includes files in tree",
+	},
+	cli.IntFlag{
+		Name:  "depth, d",
+		Usage: "sets the depth threshold",
+		Value: -1,
+	},
+}
+
+// trees files and folders.
+var treeCmd = cli.Command{
+	Name:   "tree",
+	Usage:  "list buckets and objects in a tree format",
+	Action: mainTree,
+	Before: setGlobalsFromContext,
+	Flags:  append(treeFlags, globalFlags...),
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} [FLAGS] TARGET [TARGET ...]
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+   1. List buckets on Amazon S3 cloud storage in a tree format.
+      $ {{.HelpName}} myS3
```

Comment:
We don't have this type of convention in other commands why can't we just say `s3/` ? Also we should provide some MinIO examples as well showing `myminio`

Reply:
@harshavardhana ,
That was my comment and Praveen changed it per my request. It thought it would be more clear, but if the convention is the other way, then we should use `s3/`.

Reply:
Because we already have a default alias for `s3/`  we can just use that.  Also, we have started moving to provide `myminio` in our examples.

Reply:
changed it to s3/ :+1: